src/components/DashboardEditor/editorUtils.jsx

```diff
@@ -451,8 +442,53 @@ export const handleDataSeriesChange = (
         ...cardConfig,
         content: { ...cardConfig.content, series },
       };
-    case CARD_TYPES.IMAGE:
-      dataSection = [...(cardConfig.content?.hotspots || [])];
+    case CARD_TYPES.TABLE: {
```

Comment:
This is more of a general wondering, but why do we have these huge external functions that handles changes for _all_ the different cards? The handleDataSeriesChange  signature has 5 parameters and is being called with null values on different places depending on the caller.

Reply:
for sure, this is a refactoring that needs to happen, I'll see if I can get to it in a future PR